### PR TITLE
[Py3] WSGI part 2

### PR DESCRIPTION
**Pull request by MartinBasti**
with this PR:
* server can be installed with python3-mod_wsgi
* any xmlrpc test can be executed to find a new py3 issues (still a lot of them there)

### CODE REVIEW

File: ipaserver/plugins/baseldap.py [hunk 1]
```diff
@@ -1036,8 +1037,8 @@ def process_attr_options(self, entry_attrs, dn, keys, options):
                     except ValueError:
                         if isinstance(delval, bytes):
                             # This is a Binary value, base64 encode it
-                            delval = unicode(base64.b64encode(delval))
-                        raise errors.AttrValueNotFound(attr=attr, value=delval)
+                            delval = base64.b64encode(delval).decode('ascii')
+                            raise errors.AttrValueNotFound(attr=attr, value=delval)
```
Comment (tiran) [inline ipaserver/plugins/baseldap.py]: Why did you indent the line?

Comment (MartinBasti) [inline ipaserver/plugins/baseldap.py]: I have no idea

File: ipaserver/plugins/baseldap.py [hunk 2]
```diff
@@ -255,7 +255,8 @@ def entry_to_dict(entry, **options):
                 value = list(entry.raw[attr])
                 for (i, v) in enumerate(value):
                     try:
-                        value[i] = v.decode('utf-8')
+                        if isinstance(v, bytes):
+                            value[i] = v.decode('utf-8')
```
Comment (HonzaCholasta) [inline ipaserver/plugins/baseldap.py]: `entry.raw[]` should always return a list of bytes. Why the `if`?

File: ipalib/rpc.py [hunk 3]
```diff
@@ -308,7 +308,7 @@ def json_encode_binary(val, version):
             encoded = encoded.decode('ascii')
         return {'__base64__': encoded}
     elif isinstance(val, Decimal):
-        return {'__base64__': base64.b64encode(str(val))}
+        return {'__base64__': base64.b64encode(str(val).encode('ascii'))}
```
Comment (HonzaCholasta) [inline ipalib/rpc.py]: The value must be text, but `base64.b64encode()` returns binary data, so you should also decode its result.

File: ipaserver/plugins/baseldap.py (hunk 2 above)
Comment (MartinBasti) [inline ipaserver/plugins/baseldap.py]: will be fixed in ipaldap

File: ipalib/rpc.py (hunk 3 above)
Comment (MartinBasti) [inline ipalib/rpc.py]: ok

File: ipalib/rpc.py [hunk 4]
```diff
@@ -308,7 +308,8 @@ def json_encode_binary(val, version):
             encoded = encoded.decode('ascii')
         return {'__base64__': encoded}
     elif isinstance(val, Decimal):
-        return {'__base64__': base64.b64encode(str(val))}
+        return {'__base64__': base64.b64encode(
+            str(val).encode('ascii')).decode('ascii')}
```
Comment (HonzaCholasta) [inline ipalib/rpc.py]: Could we please get rid of this non-sense and return `unicode(val)`? (It is a backward compatible change.)

Comment (MartinBasti) [inline ipalib/rpc.py]: as you wish

**Comment by HonzaCholasta** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/caa560ca79e4038b161b27d11e3f144606dbbcdb
https://fedorahosted.org/freeipa/changeset/a93b2bea5ce88a934ba2ab39bdaa518fb55064c4
https://fedorahosted.org/freeipa/changeset/a3d3b0ad2537c9d11d9c6108c31e079f0dfcf31c
https://fedorahosted.org/freeipa/changeset/03d0a55e8a21a334ca4dc625527cae93633a7314
https://fedorahosted.org/freeipa/changeset/a584758cfb87567a9c640ae107903b0f6c9fec30
https://fedorahosted.org/freeipa/changeset/ab53d80883320060769b7bfada2a813b345b9e4a
https://fedorahosted.org/freeipa/changeset/4c84341b8bc14cd19a4e2c2df4c13b95ff7eeb05

**Pull request closed without merging**